pkg/storage/utils/decomposedfs/node/node.go

```diff
@@ -1173,12 +1173,24 @@ func (n *Node) FindStorageSpaceRoot() error {
 
 // MarkProcessing marks the node as being processed
 func (n *Node) MarkProcessing() error {
-	return n.SetMetadata(xattrs.StatusPrefix, ProcessingStatus)
+	return n.MetaChange(func() error { return n.SetMetadata(xattrs.StatusPrefix, ProcessingStatus) })
 }
 
 // UnmarkProcessing removes the processing flag from the node
 func (n *Node) UnmarkProcessing() error {
-	return n.RemoveMetadata(xattrs.StatusPrefix)
+	return n.MetaChange(func() error { return n.RemoveMetadata(xattrs.StatusPrefix) })
+}
+
+// MetaChange executes an operation on the node without changing the underlying files mtime
+func (n *Node) MetaChange(op func() error) error {
+	fi, err := os.Stat(n.InternalPath())
```

Comment:
I think this stat call is not needed. The provided func should (and will) be prepared to that error. Im most cases it is just another (time consuming) stat.

Reply:
The goal of `MetaChange` method is to change the xattrs of a file without changing its mtime.  That means I need to
- get current mtime and store in a variable
- do the change
- overwrite new mtime with stored one

I don't get how to get current mtime without a stat? Or maybe I got you wrong?

Reply:
I am sorry, I overlooked that. Forget it.